doc/source/ray-core/actors/terminating-actors.rst

```diff
@@ -191,3 +192,54 @@ You could see the actor is dead as a result of the user's `exit_actor()` call:
       is_detached: false
       placement_group_id: null
       repr_name: ''
+
+
+.. _actor-cleanup:
+
+Actor cleanup with `__ray_shutdown__`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When an actor terminates gracefully, Ray calls the ``__ray_shutdown__()`` method
+if it exists, allowing cleanup of resources like database connections or file handles.
+
+.. tab-set::
+
+    .. tab-item:: Python
+
+        .. testcode::
+            :skipif: True
+
+            import ray
+
+            @ray.remote
+            class DatabaseActor:
+                def __init__(self):
+                    self.db_connection = connect_to_database()
+                    
+                def __ray_shutdown__(self):
+                    if self.db_connection:
+                        self.db_connection.close()
+                
+                def query(self, sql):
+                    return self.db_connection.execute(sql)
+
+            actor = DatabaseActor.remote()
+            ray.get(actor.query.remote("SELECT * FROM users"))
+            del actor  # __ray_shutdown__() is called automatically
+
+When ``__ray_shutdown__()`` is called:
+
+- **Automatic termination**: When all actor handles go out of scope (``del actor`` or natural scope exit)
+- **Manual graceful termination**: When you call ``actor.__ray_terminate__.remote()``
+
+When ``__ray_shutdown__()`` is **NOT** called:
+
+- **Force kill**: When you use ``ray.kill(actor)`` - the actor is killed immediately without cleanup.
+- **Unexpected termination**: When the actor process crashes or exits unexpectedly (such as ``os._exit()`` or segfaults).
+
+**Important notes:**
+
+- ``__ray_shutdown__()`` runs after all actor tasks complete and the actor is idle.
```

Comment:
```suggestion
- ``__ray_shutdown__()`` runs after all actor tasks complete.
```
"the actor is idle" is not meaningful and is misleading (users could still have code running on a background thread)